modules/nlu/src/backend/module-lifecycle/on-server-started.ts

```diff
@@ -57,9 +57,10 @@ const registerMiddleware = async (bp: typeof sdk, state: NLUState) => {
           defaultLanguage,
           bp.logger.forBot(botId)
         )
-        const nluResults = await predictionHandler.predict(preview, nlu?.includedContexts)
+        const nluResults = await predictionHandler.predict(preview)
 
-        _.merge(event, { nlu: nluResults ?? {} })
+        const nlu = { ...nluResults, includedContexts: event.nlu?.includedContexts ?? [] }
```

Comment:
From what I remember, includedContexts will impact prediction as well (like we don't even try to predict ctx that are not specified in included ctx) but maybe that changed ?

Reply:
It does only have an impact when there's no ctx_model. This has not changed in any recent PR, it has been the behavior since I was here.